R/ocjr-drug-court-SQ780.R

```diff
@@ -1,17 +1,522 @@
+#     (poss & drug & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (uccs_property & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (larceny & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (receive_stolen & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (embezzle & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (misc & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (pawn & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (hospitality & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))|
+#     (forgery & !(dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife | manufacture | commercial | school_park_church | conspiracy | maintaining | minor | inmate | vehicle | proceeds))
+#   ) |>
+#   select(id, case_type, date_filed, count_as_disposed, disposition_date,
+#          disposition, charge_desc, offense_category_desc, offense_type_desc,
+#          probability, uccs_code, party, uccs_drug_plus, poss, drug,
+#          uccs_property, larceny, receive_stolen, embezzle, misc, pawn,
+#          hospitality, forgery)
+
+# This should be equivalent to the above, minus the `sq780_statute = case_when`
+# and `filter(count_as_disposed == "")` bits, but is hopefully more readable
+clean_df <- data_clean |>
+  mutate(
+    # Is the charge a drug possession charge covered by SQ780? -----------------
+    sq780_drugs = if_else(
+      # UCCS code or regex indicating drug possession, and no disqualifying regex
+      (uccs_drug_plus | regex_drugs) & !disqualifying_regex, TRUE, FALSE),
+    # Is the charge a property crime covered by SQ780? -------------------------
+    sq780_props = if_else(
+      # UCCS code or regex indicating 780 property crime, and no disqualifying regex
+      (uccs_property | regex_props) & !disqualifying_regex, TRUE, FALSE),
```

Comment:
This is just a re-written version of the `filter()` call in the chunk I commented out above. I also decided to just add these as new variables, because I want to keep the non-SQ780 charges in there as well for now and wait to filter them out llater. That's mostly just to make it easier to look at the ones that got excluded.